ui/apps/dashboard/src/components/Events/SendEventModal.tsx

```diff
@@ -50,6 +119,27 @@ export function SendEventModal({
 
   const isBranchChild = environment.type === EnvironmentType.BranchChild;
   const envName = environment.name;
+  const [tabs, setTabs] = useState(
+    buildTabs({
+      envName,
+      payload,
+      eventKey,
+      sendEventURL,
+      isBranchChild,
+      copyToClipboardAction,
+      sendEventAction,
+    })
+  );
```

Comment:
The "Send event" button is disabled until the tab is switched.

https://github.com/user-attachments/assets/9c65feda-2582-484e-9f96-f8009e0de401

Changing `tabs` to a derived state (`useMemo` instead of `useState`) seems to fix it:
```suggestion
  const tabs = useMemo(() => {
    return buildTabs({
      envName,
      payload,
      eventKey,
      sendEventURL,
      isBranchChild,
      copyToClipboardAction,
      sendEventAction,
    });
  }, [
    copyToClipboardAction,
    envName,
    eventKey,
    isBranchChild,
    payload,
    sendEventAction,
    sendEventURL,
  ]);
```